crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3838,8 +3816,10 @@ pub(crate) mod tests {
     }
 
     #[test]
-    fn typing_vs_typeshed_no_default() {
-        let db = setup_db();
+    fn typing_vs_typeshed_no_default() -> anyhow::Result<()> {
```

Comment:
@AlexWaygood previously expressed a preference for using `unwrap()` in tests, rather than having tests return a `Result` and then using `?` in the test. We have some existing tests using both styles. I don't have strong feelings, but I think I shared Alex's preference after writing tests in both styles; the need to end the test with `Ok(())` is irritating.

Do you have a preference in the other direction?

Reply:
> Do you have a preference in the other direction?

I do not. I usually also use `.unwrap()` liberally in tests. Sometimes the code looks a lot nicer if there are many `.unwrap()`s that can be replaced with `?`.

For `TestDbBuilder` specifically, I first wrote a version that didn't return a `Result`. I then changed it because I thought there might be more complex test setups in the future where it would be useful for the developer to see a good error message if they did something wrong in the setup of the test.

And note that `setup_db()` still returns a plain `TestDb`, not a `Result`. It's just the builder that returns a `Result`.

For the particular example here... nothing can go wrong when just setting the Python version, so I'll change this back to `.unwrap()` style.